contracts/rewards/StableLPMaker.sol

```diff
@@ -0,0 +1,183 @@
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/ISwap.sol";
+
+contract StableLPMaker is Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    // pTri is the contract that disburses TLP tokens 
+    // tlpToken is the trisolaris lp token address
+    // dao is the dao address that receives funds
+
+    ISwap public stableSwap;
+    address public pTri;
+    address private immutable usn;
+    address private immutable usdc;
+    address private immutable usdt;
+    address private immutable tlpToken;
+    address public dao;
+
+    uint256 public polPercent; 
+
+    event LogSetpTri(address oldpTri, address newpTri);
+    event LogSetdao(address oldDao, address newDao);
+    event LogSetStableSwap(address oldStableSwap, address newStableSwap);
+    event LogProtocolOwnedLiquidity(uint256 oldpolPercent, uint256 newStableSwap);
+
+    event LogWithdrawFees();
+
+    event LogSwapStableTokenToUsn(address stableTokenToConvertToUsn, uint256 stableTokenAmount);
+    event LogAddliquidityToStableSwap(uint256 usnAmount);
+
+    event LogLpTokensSentTopTRI(uint256 tlpAmount);
+    event LogLpTokensSentToDao(uint256 daoAmount);
+
+    constructor(
+        address _stableSwap,
+        address _pTri,
+        address _usn,
+        address _usdt,
+        address _usdc ,
+        address _tlpToken,
+        address _dao
+    ) public {
+        stableSwap = ISwap(_stableSwap);
+        pTri = _pTri;
+        usn = _usn;
+        usdt = _usdt;
+        usdc = _usdc;
+        tlpToken = _tlpToken;
+        dao = _dao;
+    }
+
+    // C6: It's not a fool proof solution, but it prevents flash loans, so here it's ok to use tx.origin
+    modifier onlyEOA() {
+        // Try to make flash-loan exploit harder to do by only allowing externally owned addresses.
+        require(msg.sender == tx.origin, "StableLPMaker: must use EOA");
+        _;
+    }
+
+    function withdrawStableTokenFees() public onlyEOA {
+        // Withdraw admin fees from the Stableswap Pool to stable tokens
+        stableSwap.withdrawAdminFees();
+
+        emit LogWithdrawFees();
+    }
+
+    // This onyl works for USDC and USDT. If we have more tokens, need to make this more dynamic
+    function swapStableTokensToUsn() public onlyEOA {
+        address[2] memory stableTokensToSwapToUsn = [usdc, usdt];
+        for (uint256 i = 0; i < stableTokensToSwapToUsn.length; i++) {
+            IERC20(stableTokensToSwapToUsn[i]).approve(
+                address(stableSwap),
+                IERC20(stableTokensToSwapToUsn[i]).balanceOf(address(this))
+            );
+
+            uint256 stableTokenAmount = IERC20(stableTokensToSwapToUsn[i]).balanceOf(address(this));
+
+            stableSwap.swap(
+                stableSwap.getTokenIndex(stableTokensToSwapToUsn[i]),
+                stableSwap.getTokenIndex(usn),
+                stableTokenAmount,
```

Comment:
in the wild it might make sense to add some buffer to this quantity
like `minAmount = 999/1000*stableTokenAmount`

Reply:
something like this ?
uint256 minAmouunt = 999.div(1000).mul(stableTokenAmount)